core/lib/multivm/src/versions/vm_latest/tests/l1_l2_complete_tx_execution.rs

```diff
@@ -0,0 +1,602 @@
+
+    let l2_system_contracts_upgrade_tx_hash = H256::zero(); // Dummy 0x000...000
+    let l2_system_contracts_upgrade_batch_number = 5u64.into(); // Dummy value
+
+    // Generate dummy batch hashes (bytes32) with the correct length
+    let num_hashes = total_batches_committed.as_u64() - total_batches_executed.as_u64() + 1;
+    let batch_hashes: Vec<H256> = (0..num_hashes).map(|_| H256::zero()).collect(); // Dummy 0x000...000
+
+    // Constructing the priority tree commitment
+    let next_leaf_index = 30u64.into(); // Dummy value
+    let start_index = 25u64.into(); // Dummy value
+    let unprocessed_index = 28u64.into(); // Dummy value
+
+    // Generate dummy sides (bytes32) with arbitrary length
+    let sides: Vec<H256> = (0..3).map(|_| H256::zero()).collect(); // Dummy 0x000...000
+
+    // Step 2: Convert the PriorityTreeCommitment to Tokens
+    let priority_tree_commitment = Token::Tuple(vec![
+        Token::Uint(next_leaf_index),
+        Token::Uint(start_index),
+        Token::Uint(unprocessed_index),
+        Token::Array(
+            sides
+                .iter()
+                .map(|hash| Token::FixedBytes(hash.as_bytes().to_vec()))
+                .collect(),
+        ),
+    ]);
+
+    // Step 3: Convert the HyperchainCommitment to Tokens
+    let hyperchain_commitment = Token::Tuple(vec![
+        Token::Uint(total_batches_executed),
+        Token::Uint(total_batches_verified),
+        Token::Uint(total_batches_committed),
+        Token::FixedBytes(l2_system_contracts_upgrade_tx_hash.as_bytes().to_vec()),
+        Token::Uint(l2_system_contracts_upgrade_batch_number),
+        Token::Array(
+            batch_hashes
+                .iter()
+                .map(|hash| Token::FixedBytes(hash.as_bytes().to_vec()))
+                .collect(),
+        ),
+        priority_tree_commitment,
+    ]);
+
+    // Step 4: Encode the data
+    let chain_mint_data = encode(&[hyperchain_commitment]);
+
+    // Encode the data
+    let encoded_bridgehub_mint_data = encode(&[
+        chain_id.clone(),
+        Token::Bytes(stm_data),
+        Token::Bytes(chain_mint_data),
+    ]);
+
+    // Define the L1_CHAIN_ID, sender, and _additionalData
+    let l1_chain_id = Token::Uint(0.into()); // Chain ID hasn't been set
+    const OFFSET: H160 = H160([
+        0x11, 0x11, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x11, 0x11,
+    ]);
+
+    fn sub_h160(lhs: H160, rhs: H160) -> H160 {
+        let mut result = [0u8; 20];
+        let mut borrow = 0u8;
+
+        for (i, (a, b)) in lhs.0.iter().zip(rhs.0.iter()).enumerate().rev() {
+            let (sub, overflow) = a.overflowing_sub(*b);
+            let (sub_with_borrow, borrow_overflow) = sub.overflowing_sub(borrow);
+            result[i] = sub_with_borrow;
+            borrow = (overflow || borrow_overflow) as u8;
+        }
+
+        H160(result)
+    }
+
+    fn undo_l1_to_l2_alias(l2_address: H160) -> H160 {
```

Comment:
seems like a good utility function, could you please move it to the utils folder